cdap-app-fabric/src/main/java/co/cask/cdap/security/TokenSecureStoreUpdater.java

```diff
@@ -231,8 +238,36 @@ public long getUpdateInterval() {
 
   @Override
   public SecureStore update(String application, RunId runId) {
+    try {
+      if (Constants.Service.MASTER_SERVICES.equals(application)) {
+        // don't impersonate for the master.services Twill app
+        return doUpdate();
+      }
+
+      ProgramId programId = TwillAppNaming.fromTwillAppName(application);
```

Comment:
Yea, we just need the NamespaceId in this usage of `fromTwillAppName`. However, other places we need the entire ProgramId, so left this method to parse the entire thing.